app/src/main/java/com/github/se/polyfit/ui/components/button/PrimaryButton.kt

```diff
@@ -24,8 +28,15 @@ fun PrimaryButton(
     buttonShape: RoundedCornerShape = RoundedCornerShape(50.dp),
     icon: @Composable (() -> Unit)? = null
 ) {
+  var pressed by remember { mutableStateOf(false) }
+
   Button(
-      onClick = onClick,
+      onClick = {
+        if (!pressed) {
+          onClick()
+          pressed = true
+        }
```

Comment:
Looks good, maybe pressed = true before onClick() but not important

Reply:
Nice idea, fixed